### ISSUE TITLE
Improve checks for hardcoded solutions

### ISSUE THREAD

**Issue by jelaniwoods** [appdev-projects/float-chapter#7]
To something like:

```rb
if !line[0] =="#" || line[0] == "p" || line[0] == "puts"
```

instead of `include?`. This should be more accurate b/c p could appear in a comment (w/ the word `# Example` for instance) which would prevent the test from ignoring that line.

There is the case when `p` could be the first letter in a variable, not using the `p` method, but that should also be safe to ignore since it's not a hard coded solution.

**Comment by raghubetina** [comment]
Maybe this won't be an issue but what if the `#` wasn't the first letter in the line? E.g.

```
# Your output should look like:
    # 5.3
```

Maybe a regex would do the trick?

**Comment by jelaniwoods** [comment]
Yup, I think regex is the way. I just wanted to write something down like "check that first letter is not `#` or first letter is `p`" somewhere before I forgot since I'd need to test Regex some more.

**Comment by raghubetina** [comment]
I’m trying to think of a regex that would cover all cases. In english, “If the first non-whitespace character(s) in a line is 'p' or 'puts' , then the rest of that line can’t contain '5.3'”. Would that do it?

**Comment by jelaniwoods** [comment]
> I’m trying to think of a regex that would cover all cases. In english, “If the first non-whitespace character(s) in a line is 'p' or 'puts' , then the rest of that line can’t contain '5.3'”. Would that do it?

I think so. We could maybe use `(^\s*)(puts|p)` and check the second capture group.

```rb
      if line.scan(/(^\s*)(puts|p)/)[0]&.last
        expect(line).to_not match(/5.3/),
          "Expected 'float_find_hypotenuse.rb' to NOT literally print '5.3', but did anyway."
      end
```
which isn't super readable, but I think does the right thing.
